src/external/ocaml-rocksdb/rocks_linker_flags_gen.ml

```diff
@@ -14,9 +14,11 @@ let () =
         ; "-lc++abi"
         ; "-lc++" ]
     | "Linux" ->
-        [ "-Wl,--push-state,-whole-archive"
+        [ sprintf "-L%s" cwd
+        ; "-fuse-ld=lld"
+        ; "-Wl,--whole-archive"
         ; "-lrocksdb_stubs"
-        ; "-Wl,--pop-state"
+        ; "-Wl,--no-whole-archive"
```

Comment:
Can you explain the differences between push/pop state and the whole/no-whole archive here? What are the benefits of this new configuration?